### DISCUSSION TITLE
DCM estimates donot match with data

### DISCUSSION THREAD

**Discussion by anudeep-uab** [bachlab/PsPM#783] [Q&A] [answered]
Dr. Bach, 
I am a student at the University of Alabama at Birmingham, working in Dr. Knight's lab. I am analyzing SCR data from a fear conditioning study using the DCM model in PSPM (ver. 5.0.0). The conditioning procedure includes pseudorandomized presentations of CS+ (25 trials), CS- (25 trials), and UCS (i.e., unpaired, 25 trials). Each conditioned stimulus (CS+/CS-) was presented for 8 seconds, and the UCS was presented 500 ms before the offset of the CS+. In addition to the conditioned/flexible responses to CS+ and CS-, I am also interested in looking at the unconditioned responses/fixed responses to paired (i.e., CS+UCS) and unpaired UCS presentations.

I have been looking at per-trial estimates from DCM for a participant, and I noticed that the amplitude estimates for the fixed responses, specifically for the UCS paired with the CS+, were exactly the same value for multiple trials. In addition, it looks like it may be identifying UCRs and assigning a large amplitude even when a UCR does not appear to exist. That is, on visual inspection, this participant doesn’t look very responsive. I am not sure how this could happen and am concerned that I might have made a mistake setting up the model.  Could you please help me understand what is going on with this participant data. I am attaching the files in the hope it will help evaluate the issue. Kindly let me know if you need any further information. Thank you. 

Attachments: https://uab.box.com/s/mle0s2zh50e6kh1fx3glq6cca04dvgv7
Password: Pspm5dcm

Anudeep

**Comment by dominikbach** [comment]
Hi Anudeep

thanks for reaching out and providing the material to reproduce the problem.

I'm fairly sure the root cause of the issue is the bidirectional highpass filter in combination with rare and quite pronounced SCR. This causes filter ringing which in turn affects the "apparent baseline" of the data which is created by subtracting the mean. You can appreciate this by loading the model into the workspace and plotting the filtered data:

`plot(dcm.input.scr{1}); `

In a nutshell, in a given trial without actual SCR, responses are modelled to "keep the signal high" but because there is little temporal variation, the parameters cannot be estimated precisely and so the VBA algorithm sticks with the priors, which is why values are precisely repeated across trials. 

See this related issue here: https://github.com/bachlab/PsPM/discussions/765

I'd suggest you revert to unidirectional filtering. Let me know if problems persist.

Dominik

**Comment by anudeep-uab** [comment]
Dr. Bach, 
 
Thanks for your help explaining the problem. We changed the filter settings to unidirectional and reanalyzed the data. Although the values have changed some, we still see similar repeated values for fixed responses for the same CS+UCS trials as before. We repeated the analysis with the unidirectional filter for other participants (with more frequent SCRs) and also see the same issue. Below, you can find the Matlab input and output data structures for the previous participant and a different participant (with more frequent SCRs) analyzed using DCM with unidirectional filter setting. Please let me know if you need any further information. Thanks for your help. 
 
Attachments: https://uab.box.com/s/o3oejy69ugrc2hcuf5hr0tr4yjtdg47k
Password: Pspm5dcm
 
Anudeep

**Comment by dominikbach** [comment, marked as answer]
Hi Anudeep

since the bidirectional filter isn't the root cause, I investigated further:

1. I verified that indeed the constant values already appear in the raw VBA output, indicating that the algorithm sticks with the prior mean (presumably due to - loosely speaking - lack of sufficient information in the data). You can verify e.g. for participant 07, trial 5-7 with `dcm.sn{1}.posterior(trl).muTheta(14)`, where `14` is the index of the first fixed response amplitude for this particular timing definition. You can also verify that this corresponds to the values in `dcm.stats` with `exp(dcm.sn{1}.posterior(trl).muTheta(14))/dcm.sn{1}.eSCR_unit*dcm.sn{1}.newzfactor`

2. I noticed something unusual in the timing definition. For some of the trials (not the same ones as the offending UR estimates), the time window for the CR starts with `-1` and ends with the US time point. Is this intended? (There is a possibility to use negative onset values for events that should be ignored, but this only works if both ends of the interval are negative. Also, this feature has not been extensively tested.) I assume that the algorithm will model a flexible response over the entire time from experiment start to the US of that trial, and this might absorb other responses that happen during this interval.
